src/index.js

```diff
@@ -112,7 +116,12 @@ function addMetadata({ types: t }) {
           ),
         ]);
 
-        const { root, filename } = state.file.opts;
+        let { root, filename } = state.file.opts;
+
+        if (hasEmbroiderPrefix(filename)) {
+          filename = getEmbroiderStrippedPrefixPath(filename);
+        }
+
         const relativeFilePath = path.relative(root, filename);
```

Comment:
Rather than do this, I would instead encapsulate the entire logic for getting a filename in a separate util, and test that with all the correct permutations. This does a nicer job encapsulating the logic in a single location, whereas this spreads it out within the call site.